scripts/gen-auth-api-spec.ts

```diff
@@ -130,6 +128,70 @@ async function toOpenapi3(discovery: any): Promise<any> {
   return openapi3;
 }
 
+const pathParamsForFlatPathParam: Map<string, string> = new Map([
+  ["{projectsId}", "{targetProjectId}"],
+  ["{tenantsId}", "{tenantId}"],
+]);
+
+const paramPattern = /{([^}]+)}/g;
+
+function replaceWithFlatPath(discovery: { [key: string]: any }): void {
```

Comment:
Is it possible to type the discovery more narrowly than `any`? (I'm not super familiar with this code)

Reply:
Most of the objects from the schema won't have good typing since they are a Google-specific format. You can however write your own interfaces, if that's worth your time, on a case by case basis.